### PR TITLE
remove jrep from Interval

### PR DESCRIPTION
**Pull request by cseed**
remove _convert_to_j, no longer used (pending some other PRs)

note, this is a breaking change, I'm removing overlaps and contains from the Interval interface.  I doubt anyone is using them.  Users should use hl.eval and the corresponding functions on IntervalExpression.  FYI @tpoterba for release notes (or pushback)

### CODE REVIEW

**Comment by jigold** [comment]
You're also removing Interval.point_type (just to be clear what the breaking changes are).

**Comment by cseed** [comment]
Good catch, thanks!

**Comment by cseed** [comment]
FYI, this won't pass until https://github.com/hail-is/hail/pull/5151 goes in (eliminating the last use of _convert_to_j).

**Comment by tpoterba** [comment]
I'm not sure what to think about the breaking change. Probably nobody is using it, but it still is a violation of our promise.

**Comment by tpoterba** [comment]
we could add overlaps and contains written in Python.

**Comment by tpoterba** [comment]
(it wouldn't be trivial since we'd need to write orderings)

**Comment by cseed** [comment]
I'm not either.  It's clearly the right thing.  We can write them in terms of `hl.eval`, but I don't think they will work quite the same since we don't have the point type and type inference could fail (e.g. the endpoints are None).

**Comment by tpoterba** [comment]
why can't we store the point type?

**Comment by cseed** [comment]
I'll put point_type back in for the sake of compatibility.  This will depend on https://github.com/hail-is/hail/pull/5138 so I'm waiting for that to go in first.

File: hail/python/hail/utils/interval.py [hunk 1]
```diff
@@ -28,50 +28,33 @@ class Interval(object):
...
-            includes_end)
-
     def __str__(self):
-        return self._jrep.toString()
+        if isinstance(self._start, Locus) and self._start.contig == self._end.contig:
+            bounds = f'{self._start}-{self._end.position}'
+        else:
+            bounds = f'{self._start}-{self._end}'
+        open = '[' if self._includes_start else '('
+        close = '[' if self._includes_end else '('
```
Comment (jigold) [inline hail/python/hail/utils/interval.py]: These should be the opposite characters. I.e. `]` and `)`

**Comment by cseed** [comment]
I put point_type back.  Should be ready to go now.  Also, fixed the close definition, good catch.

I also removed an additional use of _convert_to_j in import_bgen to get the tests to pass from this PR (sorry my stacking got a bit mixed up): https://github.com/hail-is/hail/pull/5150/files#diff-36d21c1427efe06a781cd36ef5aa8678R961.  You can also wait for that to go in and I'll rebase if you're worried about the change.

Finally, the imports are a bit of a mess since I wanted to use hail_type in interval.py which is also imported by the types and expr files.  @tpoterba I think we should remove types from expr and remove java from utils (we're confusing user utils like hadoop_* and Interval from internal utils like Env and java stuff which don't seem related) and have a clear "import" graph: javautils > types > utils > expr.

**Comment by tpoterba** [comment]
`clear "import" graph: javautils > types > utils > expr` 

Sounds good to me.

**Comment by cseed** [comment]
https://github.com/hail-is/hail/pull/5150 went in and I rebased, so the diff should be clean now.

File: hail/python/hail/utils/interval.py [hunk 2]
```diff
@@ -26,52 +27,45 @@ class Interval(object):
...
-    def __init__(self, start, end, includes_start=True, includes_end=False):
-        from hail.expr.expressions import impute_type, unify_types_limited
-        start_type = impute_type(start)
-        end_type = impute_type(end)
-        point_type = unify_types_limited(start_type, end_type)
-
+                      includes_end=bool,
+                      point_type=nullable(hail_type))
+    def __init__(self, start, end, includes_start=True, includes_end=False, point_type=None):
         if point_type is None:
```
Comment (jigold) [inline hail/python/hail/utils/interval.py]: Why did you change the signature of the constructor to take `point_type`? Are you trying to eliminate the checks if the user passes what the point_type is?

Comment (tpoterba) [reply hail/python/hail/utils/interval.py]: This is to preserve the interface - we need to be able to construct intervals in python that might contain None, and type imputation will fail on None.

(I think)

Comment (cseed) [reply hail/python/hail/utils/interval.py]: Yes, what Tim said.  I'm removing Interval._from_java, which took took the point type instead of inferring it.  The calls to _from_java turned into calls to the constructor, so I need to pass in the point type in those cases to preserve the old behavior.  I think adding an argument with a default isn't a breaking change.  I could have an alternate constructor in this case but I think this is clearer.
